std/stdio.d

```diff
@@ -2808,6 +2808,89 @@ is empty, throws an `Exception`. In case of an I/O error throws
         assert(i == witness.length);
     }
 
+/**
+Like `asCharsRange` but returns a range o `ubyte`s, not `char`s.
+*/
+    auto asUbytesRange(size_t bufferSize = 4096)
+    {
+        assert(bufferSize > 0);
+
+        import std.algorithm.iteration;
+        return byChunk(bufferSize).joiner;
+    }
+
+/**
+Returns an $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
+set up to read from the file handle.
+
+The element type for the range will be `char`. Range primitives
+may throw `StdioException` on I/O error.
+
+Example:
+---------
+void main()
+{
+    // Read standard input
+    foreach (char c; stdin.asCharsRange)
+    {
+        ... use c ...
+    }
+}
+---------
+
+The parameter may be a number (as shown in the example above) recommending the
+size of the buffer.
+
+Example:
+---------
+void main()
+{
+    // Read standard input
+    foreach (char c; stdin.asCharsRange(4096))
+    {
+        ... use c ...
+    }
+}
+---------
+
+With the mentioned limitations, `asCharsRange` works with any algorithm
+compatible with input ranges.
+
+Returns: A range initialized with the `File` object.
+
+Asserts: Buffer size is non-zero. In case of an I/O error throws
+`StdioException`.
+ */
+    auto asCharsRange(size_t bufferSize = 4096)
+    {
+        import std.algorithm;
+        return map!(b => cast(char) b)(asUbytesRange(bufferSize)); // TODO: efficient?
```

Comment:
What you probably want here is `byChunk(bufferSize).map!assumeUTF.joiner`, but the implicit non-validating conversion issue stands.